### PR TITLE
feat(components): [el-select-v2] mouse trigger scrollbar show/hide

### PR DESCRIPTION
**Pull request by btea**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by msidolphin** [comment]
Maybe it's better to use css  stylecontrol?

**Comment by JeremyWuuuuu** [comment]
> Maybe it's better to use css style control? cc @JeremyWuuuuu WDYT

Nope, since the visibility is controlled via `visible` for `virtualized scrollbar` it should be managed by `JavaScript` IMHO.

We should also provide a props for `always-on`.

**Comment by JeremyWuuuuu** [comment]
In this PR we have partially implemented the functionality but it should be more accurate for both `naming` and `behavior`

As I mentioned about

1. The name should be `scrollbarVisible` or something else.
2. We need a flag for `always-on` to indicates the scrollbar should be always displayed.

**Comment by msidolphin** [comment]
> > Maybe it's better to use css style control? cc @JeremyWuuuuu WDYT
> 
> Nope, since the visibility is controlled via `visible` for `virtualized scrollbar` it should be managed by `JavaScript` IMHO.
> 
> We should also provide a props for `always-on`.

I see that scrollVisible is controlled by virtual-list, I think it's better to use `el-vl__wrapper:hover`, just like el-scrollbar

**Comment by btea** [comment]
So, which way is better?  
Maybe it's better to be consistent with select ?

**Comment by JeremyWuuuuu** [comment]
@btea From my point of view, I think using css to control would be easier though.

Also we need to add the `always-on` flag for our upper layer components to determine whether show the scrollbar or not.

**Comment by btea** [comment]
@JeremyWuuuuu If  use css to control, Is the code here unnecessary ?

https://github.com/element-plus/element-plus/blob/ac580d6fbd33e630c22ed3df16db3056fe88c2b0/packages/components/virtual-list/src/components/scrollbar.ts#L57

**Comment by JeremyWuuuuu** [comment]
> @JeremyWuuuuu If use css to control, Is the code here unnecessary ?
> 
> https://github.com/element-plus/element-plus/blob/ac580d6fbd33e630c22ed3df16db3056fe88c2b0/packages/components/virtual-list/src/components/scrollbar.ts#L57

Yes, we can move all the code for visibility control to `scss` file with class to control.

**Comment by btea** [comment]
@JeremyWuuuuu I see. About you said  the `always-on` flag, If the value set `true`,  `el-vl__wrapper` will add a class that Include pseudo class `hover` can control `scrollbar` show or not, otherwise, the scrollbar will never be displayed.  

Am I right?

**Comment by JeremyWuuuuu** [comment]
> @JeremyWuuuuu I see. About you said the `always-on` flag, If the value set `true`, `el-vl__wrapper` will add a class that Include pseudo class `hover` can control `scrollbar` show or not, otherwise, the scrollbar will never be displayed.
> 
> Am I right?

No, `always-on` meaning the no matter your mouse hovers the list or not, the scrollbar will always be showed, if you are a MacOS user with a mouse you can set that in your system settings, meaning that when `always-on` is absent, we should allow the mouse hover behavior to take control, when it was set to `true`, we should always display the scrollbar no matter mouse hovering or not.

**Comment by btea** [comment]
OK, I see.  Are any of you making changes  to this feature? If not, I will try it.

**Comment by JeremyWuuuuu** [comment]
> OK, I see. Are any of you making changes to this feature? If not, I will try it.

That'd be fantastic! Thank you! :bow:

**Comment by btea** [comment]
@JeremyWuuuuu Please review it.

**Comment by msidolphin** [comment]
> @JeremyWuuuuu Maybe I need to add a test? And modify the document?

The problem is not only in the lack of test, I left some comments you did not see?

**Comment by btea** [comment]
> > @JeremyWuuuuu Maybe I need to add a test? And modify the document?
> 
> The problem is not only in the lack of test, I left some comments you did not see?

Sorry，I ignored these.  I will adjust the code.

**Comment by msidolphin** [comment]
> > I see that scrollVisible is controlled by virtual-list, I think it's better to use `el-vl__wrapper:hover`, just like el-scrollbar
> 
> @msidolphin I read these comments again, About you say that `like el-scrollbar`, I found its implementation by change the `visible` to control `v-show`. This does not seem to be in line with the above use `el-vl__wrapper:hover`.
> 
> https://github.com/element-plus/element-plus/blob/a75cdde89580decdf1410ac322ae26a29ceca182/packages/components/scrollbar/src/bar.vue#L3-L6
> 
> 
> So, I'm confused now. Maybe I didn't get your point. Can you describe it in more detail? Thanks!

My point is refer to the transition effect of the scrollbar

```css
transition: var(--el-transition-duration) background-color
```

File: packages/components/select-v2/src/defaults.ts [hunk 1]
```diff
@@ -79,6 +79,10 @@ export const SelectProps = {
     type: String,
     default: 'value',
   },
+  alwaysOn: {
```
[the remark this answers is not in the document]
Comment (btea) [reply packages/components/select-v2/src/defaults.ts:82]: Maybe it's too long?

File: packages/theme-chalk/src/virtual-list.scss [hunk 2]
```diff
@@ -1,11 +1,26 @@
 @use 'mixins/mixins' as *;
 @use 'common/var' as *;
 
+@mixin showScrollbar {
+  @include b(virtual-scrollbar) {
+    display: block;
```
[the remark this answers is not in the document]
Comment (btea) [reply packages/theme-chalk/src/virtual-list.scss:6]: Do you mean to add a transition effect here?

File: packages/components/select-v2/src/defaults.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (msidolphin) [reply packages/components/select-v2/src/defaults.ts:82]: Clarity is more important

File: packages/theme-chalk/src/virtual-list.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (msidolphin) [reply packages/theme-chalk/src/virtual-list.scss:6]: Yeah, the transition of el-scrollbar like this
```css
transition: opacity 340ms ease-out
```

File: packages/components/select-v2/src/defaults.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (btea) [reply packages/components/select-v2/src/defaults.ts:82]: You are right.

File: packages/theme-chalk/src/virtual-list.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (btea) [reply packages/theme-chalk/src/virtual-list.scss:6]: Ok. I'll add it.

**Comment by msidolphin** [comment]
I think we should also add documentation and test cases

**Comment by btea** [comment]
OK, I'll add it.

**Comment by btea** [comment]
@msidolphin Please review the code.

**Review by msidolphin — APPROVED** [review]

**Pull request merged**